src/composables/useQueryParam.ts

```diff
@@ -0,0 +1,112 @@
+import { watch, Ref } from "vue";
+import { useRoute, useRouter } from "vue-router";
+
+interface QueryParamProps {
+  param: string;
+  ref: Ref;
+  valid: (T) => boolean;
+  push?: boolean;
+  convertInt?: boolean;
+  refField?: string;
+  /* eslint-disable @typescript-eslint/ban-types */
+  paramToVal?: Function;
+  /* eslint-disable @typescript-eslint/ban-types */
+  valToParam?: Function;
```

Comment:
Would something more explicit about the inputs and outputs of the function work for the typescript?

Something like the following?

`(v: string) => string`

Reply:
yeah, but we don't really know the types here is the problem.  I tried using generics `(T) => S` but it complained about that.

Reply:
ahhh i seee, tricky! alright we disable the typescript then

Reply:
just eslint, typescript is happy, just poorly informed and able to make a lot of inferences based on this typing